eng/pipelines/templates/steps/analyze.yml

```diff
@@ -43,9 +43,13 @@ steps:
     failOnStderr: true
     workingDirectory: 'sdk/${{parameters.ServiceDirectory}}'
 
-  - script: |
-      echo ensuring $(pwd)/LICENSE file exists
-      grep -L -r --include ./LICENSE -P "Copyright (\d{4}|\(c\)) Microsoft" . | tee >&2
+  - pwsh: |
+      Write-Host "ensuring $(pwd)/LICENSE file exists"
+      $patternMatches = Get-Content ./LICENSE | Select-String -Pattern 'Copyright (\d{4}|\(c\)) Microsoft'
+      if ($patternMatches.Length -eq 0) {
+        Write-Host "Could not find a LICENSE file"
```

Comment:
I think that's ideal, if there is no LICENSE at all we want to fail.

Reply:
Yes it should fail but we should make the error message useful for folks to know what is wrong instead of a random file not found error. So I'd suggest testing to make sure the message looks useful when it is missing.